### ISSUE TITLE
checkpoints_path is not saving model.save_weight

### ISSUE THREAD

**Issue by xtehol** [divamgupta/image-segmentation-keras#172]
I am following divamgupta's 'Using the python module'.

I am implementing exactly as post, but checkpoints_path is not saving any files after training. I used the command 'python main.py'. main.py is the code in 'Using the python module'.
As the model is training it prints

Epoch 1/1
10/10 [===========] ...........
('saved' , '/tmp/vgg_unet_1.model2')
('Finished Epoch' , 2)

but '/tmp/vgg_unet_1.model2' is nowhere to be found.
Does anyone know how to save the .h5 file??

**Comment by bessszilard** [comment]
Did you tried to use `model.save_weights("model.h5")` ?

I import vgg_unet the saved model with this code:

```python
from keras_segmentation.models.unet import vgg_unet

model_weight_path = 'model.h5'
model = vgg_unet(n_classes=6,  input_height=640, input_width=640)
model.load_weights(model_weight_path, by_name=True)
```

**Comment by divamgupta** [comment]
@xtehol , thanks for pointing this out. I have fixed the issue in https://github.com/divamgupta/image-segmentation-keras/commit/4c634385be4af9edcc5f98dc9b38edc12d29d97f

**Comment by Philliec459** [comment]
Hello,

I have been studying your work using this GitHub repository, blog and other written materials of yours that I have found online. Thank you. This is great work and a real help to a novice such as myself. If you do not mind I will be creating a few repositories using my interpretation of how this all might be implemented. I might be wrong, so your constructive criticism is welcome.

My first repository will be on using your pre-trained examples to interactively select the pixel value associated with a particular segment in the photograph that I am trying to isolate. I am working from your bedroom picture that you furnished in this repository. After isolating the item that I am interested in, I then create another image with just that feature and a green rectangle around it. Nothing fancy.

https://github.com/Philliec459/Basic-image-segmentation-Keras-per-work-of-Divam-Gupta <https://github.com/Philliec459/Basic-image-segmentation-Keras-per-work-of-Divam-Gupta>

The second repository will be using the kMean clusters as presented by Vidhya, repeating the same type of segment discrimination.

https://github.com/Philliec459/Basic-image-segmentation-Keras-per-Divam-Gupta-with-kMean-clusters-as-presented-by-Vidhya <https://github.com/Philliec459/Basic-image-segmentation-Keras-per-Divam-Gupta-with-kMean-clusters-as-presented-by-Vidhya>

My primary goal has been to categorize the different types of grains in a clastic petrographic thin section. This is work in progress. The next step is to estimate the Petrophysical Rock Types and Petrophysical properties from any clastic Thin Section photomicrograph. I will probably have something out on GitHub in a week or so. I will not be able to release my training set since it belongs to various oil companies, but I will release the .h5 file. 

Before I publish this in GitHub I want to be able to use augmentation which I have done in other examples, but I am not sure how to implement it using your code examples

Thank you again for your fine work. 

By the way I am from Pittsburgh originally, but left Pittsburgh in 1977 to get into the oil industry that had us working internationally for nearly 30 years. 



Best Regards, 
 
_______________________________
 
E. Craig Phillips
CEO and Chief Petrophysicist
Crested Butte Petrophysical Consultants
 







459 Cisneros Lane
Crested Butte, CO 81224
USA
Office: +1 970-343-0730
Mobile: +1 970-343-0730
email: craig@cbpetro.com
Website: www.cbpetro.com
GitHub: https://github.com/Philliec459



The information contained in this electronic message is confidential, it constitutes a profesional and/or industrial secret in terms of the current legislation, and is intended for its recipient only. If you receive this message by mistake or if you are not the recipient thereof, please notify the sender and destroy it.

La informacion contenida en este mensaje de datos es confidencial, constituye un secreto industrial y/o profesional en terminos de la legislacion vigente y se encuentra dirigida exclusivamente al destinatario indicado en dicho mensaje. Si usted recibe esta informacion por error o si usted no es el destinatario del mensaje, favor de notificar al emisor, y destruyalo.

On Mar 11, 2020, at 8:59 AM, Divam Gupta <notifications@github.com> wrote:


Closed #172 <https://github.com/divamgupta/image-segmentation-keras/issues/172>.

**Comment by divamgupta** [comment]
Augmentation can be used if you are on the master version of the repo. You can set  do_augment=True while calling the train function via the python API.

**Comment by Philliec459** [comment]
Thank you. 


The following is how I implemented your suggestion on utilizing augmentation. 


model.train(
    train_images =  "dataset1/images_prepped_train/",
    train_annotations = "dataset1/annotations_prepped_train/",
    val_images="dataset1/images_prepped_test/" ,
    val_annotations="dataset1/annotations_prepped_test/",
    verify_dataset=True,
    optimizer_name='adadelta' , do_augment=True , augmentation_name="aug_all",    
    checkpoints_path = "weights/vgg_unet_1" , epochs=5
)



I am not sure if the master version handles this internally or if I should have actually created my own augmentation subdirectory as shown below.


data_gen_args = dict(rotation_range=0.2,
                    width_shift_range=0.05,
                    height_shift_range=0.05,
                    shear_range=0.05,
                    zoom_range=0.05,
                    horizontal_flip=True,
                    fill_mode='nearest')

myGenerator = trainGenerator(20, 'data/train’, 'image’, 'label’, data_gen_args, save_to_dir = "data/train/aug”)







Best Regards, 
 
_______________________________
 
E. Craig Phillips
CEO and Chief Petrophysicist
Crested Butte Petrophysical Consultants
 







459 Cisneros Lane
Crested Butte, CO 81224
USA
Office: +1 970-343-0730
Mobile: +1 970-343-0730
email: craig@cbpetro.com
Website: www.cbpetro.com
GitHub: https://github.com/Philliec459



The information contained in this electronic message is confidential, it constitutes a profesional and/or industrial secret in terms of the current legislation, and is intended for its recipient only. If you receive this message by mistake or if you are not the recipient thereof, please notify the sender and destroy it.

La informacion contenida en este mensaje de datos es confidencial, constituye un secreto industrial y/o profesional en terminos de la legislacion vigente y se encuentra dirigida exclusivamente al destinatario indicado en dicho mensaje. Si usted recibe esta informacion por error o si usted no es el destinatario del mensaje, favor de notificar al emisor, y destruyalo.

On Mar 11, 2020, at 11:56 AM, Divam Gupta <notifications@github.com> wrote:


Augmentation can be used if you are on the master version of the repo. You can set do_augment=True while calling the train function via the python API.
